tests/system/general/test_45_FilterSolutions/test_distributed_filter/test_distributed_topology.py

```diff
@@ -0,0 +1,541 @@
+        assert design.topology.circuit_response().splitlines() == read_resource_file(
+            "ring_resonator_end_gap_extension.ckt", "Distributed"
+        )
+
+    def test_tuning_type_1(self):
+        design = ansys.aedt.core.filtersolutions.DistributedDesign(config["desktopVersion"])
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.tuning_type_1 = True
+        assert info.value.args[0] == "The Tuning Type property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.HAIRPIN
+        assert design.topology.tuning_type_1
+        design.topology.tuning_type_1 = False
+        assert design.topology.tuning_type_1 is False
+        assert design.topology.circuit_response().splitlines() == read_resource_file("tuning_type_1.ckt", "Distributed")
+
+    def test_tuning_type_2(self):
+        design = ansys.aedt.core.filtersolutions.DistributedDesign(config["desktopVersion"])
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.tuning_type_1 = True
+        assert info.value.args[0] == "The Tuning Type property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.HAIRPIN
+        assert design.topology.tuning_type_1
+        assert design.topology.circuit_response().splitlines() == read_resource_file("tuning_type_2.ckt", "Distributed")
+
+    def test_tap_position(self):
+        design = ansys.aedt.core.filtersolutions.DistributedDesign(config["desktopVersion"])
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.tap_position = TapPosition.AUTO
+        assert info.value.args[0] == "The Tap Position property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.MINIATURE_HAIRPIN
+        assert design.topology.tap_position == TapPosition.AUTO
+        design.topology.tap_position = TapPosition.SIDES
+        assert design.topology.tap_position == TapPosition.SIDES
+        assert design.topology.circuit_response().splitlines() == read_resource_file("tap_position.ckt", "Distributed")
+
+    def test_wide_band(self):
+        design = ansys.aedt.core.filtersolutions.DistributedDesign(config["desktopVersion"])
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.wide_band = True
+        assert info.value.args[0] == "The Wide Band property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.INTERDIGITAL
+        assert design.topology.wide_band is False
+        design.topology.wide_band = True
+        design.topology.pinned = True
```

Comment:
I'm not sure why pinned is being enabled here. If there is a reason, a comment would be helpful.